frontend/lib/qr_code_scanner/qr_code_scanner.dart

```diff
@@ -180,41 +180,34 @@ class _QRViewState extends State<QRCodeScanner> {
     });
   }
 
-  void _tryParseCodeContent(String codeContent) {
-    try {
-      final cardDetails = parseQRCodeContent(codeContent);
-      // TODO this method gets called multiple times in a row for some reason
-      // The following code should be exclusive, but dart does not natively
-      // support mutexes, synchronized, etc.
-      if (isDone) {
-        return;
-      }
-      isDone = true;
-      Provider.of<CardDetailsModel>(context, listen: false)
-          .setCardDetails(cardDetails);
-      Navigator.of(context).maybePop();
-    } on Exception catch (e) {
+  void _tryParseCodeContent(String codeContent) async {
+    var wasSuccessful = false;
+    // needed because this method gets called multiple times in a row after one
+    // qr code gets detected, therefore we need to protect it
+    if (isDone) {
+      return;
+    }
+    isDone = true;
+    await qrCodeParsingMutex.protect(() async {
       controller.pauseCamera();
```

Comment:
Why should we pause the camera?

Reply:
Because that's what causing this method to be called like 5 times in a row. It prevents the scanner to continue scanning in the background

Reply:
But we already have the `isProcessing` check, so this should not be necessary I guess?